posts/centralized-rules-execution-server.md

```diff
@@ -0,0 +1,100 @@
+---
+authors:
+- Mallik Sambaraju
+- Amine Ouali Alami
+tags:
+- Rules
+- Kie
+- Drools
+- JHipster
+- AWS ECS
+- Spring Boot
+date: 2010-01-23T10:00:00.000Z
+title: Centralized Rules Engine on AWS ECS
+---
+
+## Introduction
+As software engineers we architect and develop many real world applications solving many business problems, and in those applications we add business logic to solve various issues. Often this business logic is difficult to implement in application code, business rules keep on changing and hard to maintain.
+
+## Why rules engine
+A rules engine may be handy in certain cases where the business logic keeps on changing, there is no easy solution to implement the logic in the code, the code becomes cluttered or may be the business analysts want an easy way to maintain the business rules. Apart from these, rules engines implement rules execution algorithms such as Rete algorithm which are more efficient than traditional if..else or switch statements.
+
+## The problem
+At one of our recent customer's location, we were tasked to resolve a problem and the application has to look into more than hundred different scenarios to come up with a result. Also the customer wanted to have a provision to maintain the rules separately as the rules can change quite often so that the main applications do not have to be redployed whenever there is a rule change.
+
+## The solution
+We decided to use a centralized rules execution engine based on Jboss Drools (Kie execution server) to make the rules engine a loosely coupled application, to maintain the rules outside of the business application, to make the rules engine available to other applications in the organization which have similar use cases.
+
+Kie execution server can be used to instantiate and execute rules via Rest, JMS or Java client side applications, it supports runtime updates to the rules.
+
+
+## Architecture
+The first version of the rules engine is designed to adhere to the customer's infrastructure, ability to use the customer's source control repository, security standards and CI/CD process. The solution is designed to auto-recover in case of rules engine restarts/crashes.
```

Comment:
```suggestion
The first version of the rules engine is designed to adhere to the customer's infrastructure, ability to use the customer's source control repository, security standards, and CI/CD process. We built a solution to auto-recover in case of rules engine restarts or crashes.
```